src/main/java/br/com/caelum/comidinhas/handler/validation/ValidationErrorHandler.java

```diff
@@ -0,0 +1,49 @@
+package br.com.caelum.comidinhas.handler.validation;
+
+import br.com.caelum.comidinhas.exception.RelationshipNotFoundException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.MessageSource;
+import org.springframework.context.i18n.LocaleContextHolder;
+import org.springframework.http.HttpStatus;
+import org.springframework.validation.*;
+import org.springframework.web.bind.MethodArgumentNotValidException;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@RestControllerAdvice
+public class ValidationErrorHandler {
+    @Autowired
+    private MessageSource messageSource;
+
+    @ResponseStatus(HttpStatus.BAD_REQUEST)
+    @ExceptionHandler(MethodArgumentNotValidException.class)
+    public ValidationErrorsOutputDto handleValidationErrorArgumentNotValid(MethodArgumentNotValidException exception) {
```

Comment:
Método não precisa ser público.

Favorecemos o uso do modificador de acesso mais restrito.

```suggestion
    ValidationErrorsOutputDto handleValidationErrorArgumentNotValid(MethodArgumentNotValidException exception) {
```